resolver/src/resolver_future.rs

```diff
@@ -106,7 +106,7 @@ impl ResolverFuture {
     ///
     /// * `config` - configuration, name_servers, etc. for the Resolver
     /// * `options` - basic lookup options for the resolver
-    pub fn new(config: ResolverConfig, options: ResolverOpts) -> Self {
+    pub fn new(config: ResolverConfig, options: ResolverOpts) -> Box<Future<Item=Self, Error=ResolveError> + Send> {
```

Comment:
Do we need to make `new() -> Box<Future>` or could we delay that to the lookup methods?